engine/src/flutter/shell/common/shell.cc

```diff
@@ -231,27 +221,43 @@ std::unique_ptr<Shell> Shell::CreateShellOnPlatformThread(
     return nullptr;
   }
 
+  PlatformView* platform_view_ptr = platform_view.get();
   // Create the rasterizer on the raster thread.
   std::promise<std::unique_ptr<Rasterizer>> rasterizer_promise;
   auto rasterizer_future = rasterizer_promise.get_future();
   std::promise<fml::TaskRunnerAffineWeakPtr<SnapshotDelegate>>
       snapshot_delegate_promise;
   auto snapshot_delegate_future = snapshot_delegate_promise.get_future();
+
+  std::promise<std::shared_ptr<impeller::Context>> impeller_context_promise;
+  auto impeller_context_future =
+      std::shared_future<std::shared_ptr<impeller::Context>>(
+          impeller_context_promise.get_future());
+
   fml::TaskRunner::RunNowOrPostTask(
       task_runners.GetRasterTaskRunner(),
       [&rasterizer_promise,  //
-       &snapshot_delegate_promise,
-       on_create_rasterizer,                                   //
-       shell = shell.get(),                                    //
-       impeller_context = platform_view->GetImpellerContext()  //
-  ]() {
+       &snapshot_delegate_promise, impeller_context_future,
+       on_create_rasterizer,  //
+       shell = shell.get()]() {
         TRACE_EVENT0("flutter", "ShellSetupGPUSubsystem");
         std::unique_ptr<Rasterizer> rasterizer(on_create_rasterizer(*shell));
-        rasterizer->SetImpellerContext(impeller_context);
+        rasterizer->SetImpellerContext(impeller_context_future);
         snapshot_delegate_promise.set_value(rasterizer->GetSnapshotDelegate());
         rasterizer_promise.set_value(std::move(rasterizer));
       });
 
+  fml::TaskRunner::RunNowOrPostTask(
+      task_runners.GetRasterTaskRunner(),
+      fml::MakeCopyable(
+          [&,
+           impeller_context_promise = std::move(impeller_context_promise),  //
+           platform_view_ptr]() mutable {
+            TRACE_EVENT0("flutter", "CreateImpellerContext");
+            impeller_context_promise.set_value(
+                platform_view_ptr->GetImpellerContext());
```

Comment:
Add a comment explaining that GetImpellerContext lazily constructs the platform view's Impeller context and may be doing expensive initialization work.